internal/pkg/fleetapi/ack_cmd.go

```diff
@@ -101,22 +112,24 @@ func (e *AckCmd) Execute(ctx context.Context, r *AckRequest) (*AckResponse, erro
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
-		return nil, client.ExtractError(resp.Body)
+	// Read ack response always it can be sent with any status code.
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
 	}
 
-	ackResponse := &AckResponse{}
-	decoder := json.NewDecoder(resp.Body)
-	if err := decoder.Decode(ackResponse); err != nil {
+	var ackResponse AckResponse
+	if err := json.Unmarshal(body, &ackResponse); err != nil {
```

Comment:
json.Unmarshal here would pretty much only fail if the response is not parsable json. 
Calling ExtractError on error here would not help, that's why bailing out with error.
Down below, if the payload was parsable JSON, but ```if ackResponse.Action != "acks" {```, that means that it could be the JSON that contained the error information and that's when we try to ```ExtractError```.

Reply:
Awesome. Thanks for the explanation!